R-package/R/model.R

```diff
@@ -589,27 +587,20 @@ predict.MXFeedForwardModel <- function(model, X, ctx = NULL, array.batch.size =
 mx.model.load <- function(prefix, iteration) {
   symbol <- mx.symbol.load(path.expand(paste0(prefix, "-symbol.json")))
   save.dict <- mx.nd.load(path.expand(sprintf("%s-%04d.params", prefix, iteration)))
-  names <- names(save.dict)
-  arg.index <- as.integer(mx.util.filter.null(lapply(1:length(names), function(i) {
-    if (startsWith(names[[i]], "arg:")) i else NULL
-  })))
-  aux.index <- as.integer(mx.util.filter.null(lapply(1:length(names), function(i) {
-    if (startsWith(names[[i]], "aux:")) i else NULL
-  })))
+  nms <- names(save.dict)
+  
+  arg.index <- startsWith(nms, "arg:")
```

Comment:
I don't think this returns the same result with the original code.

Reply:
The current `master` code uses `lapply` to build up an integer vector of indices.  The change I am proposing uses `startsWith` itself to construct a logical vector of indices.  Up to this point the code does give a different intermediate result, but I believe through the new line 603 the end result should be the same.  The change takes advantage of the vectorization of `startsWith` and `substr` to avoid unnecessary `lapply` calls and reduce the verbosity of this section of code.